wallet/wallet.go

```diff
@@ -3760,27 +3741,23 @@ func (w *Wallet) publishTransaction(tx *wire.MsgTx) (*chainhash.Hash, error) {
 		return nil, err
 	}
 
-	var (
-		txid      = tx.TxHash()
-		returnErr error
-	)
-
+	txid := tx.TxHash()
 	_, err = chainClient.SendRawTransaction(tx, false)
 	if err == nil {
 		return &txid, nil
 	}
 
-	returnErr = MapBroadcastBackendError(err)
-
-	var errInMempool *ErrInMempool
-	var errAlreadyConfirmed *ErrAlreadyConfirmed
+	// Map the error to an RPC-specific error type.
```

Comment:
Nit: Maybe a more detailed comment that all errors are mapped to bitcoind errors no matter the used backend

Reply:
updated!